tests/iDibo_tests.py

```diff
@@ -1,11 +1,24 @@
 from nose.tools import *
-import iDibo
 
-def setup():
-    print "SETUP!"
+import unittest
+from gluon.globals import Request
 
-def teardown():
-    print "TEAR DOWN!"
+db = test_db
 
-def test_basic():
-    print "I RAN!"
\ No newline at end of file
+execfile("applications/iDibo/controllers/default.py", globals())
+db(db.auth_user.id > 0).delete()
+db.commit()
+
+
+class TestRegister(unittest.TestCase):
+    def setup(self):
+        request = Request()
+
+    def test_register_title(self):
+        self.controller.register()
```

Comment:
You have created a class "TestRegister", that is extending a "unitest.TestCase"; you are referring to that class through "self" and trying to access the "controller" of that class. However, you have never initialised a instance variable "controller" in "TestRegister" - you have only created "request": this means that the variable "controller" doesn't exist in the scope of your class.

Please have a look at:
- https://docs.python.org/2/tutorial/classes.html
- http://learnpythonthehardway.org/book/ex40.html to 45